progs/mimc.py

```diff
@@ -0,0 +1,38 @@
+from math import log, ceil
+from honeybadgermpc.mpc import PreProcessedElements, Subgroup
+
+# ROUND: iteration time of MiMC encryption function,
+# In BLS12_381, r = ceil(log(p, 3)) = 161
+ROUND = ceil(log(Subgroup.BLS12_381))
+
+
+async def mimc_mpc(context, x, k):
+    """
+    MiMC block cipher encryption encrypts message x with secret key k,
+    where x and k are elements of F_p
+    See: https://eprint.iacr.org/2016/542.pdf
+    """
+    pp_elements = PreProcessedElements()
+
+    # def cubing_share(): [x] -> [x^3]
+    async def cubing_share(x):
```

Comment:
I think this, alongside other common operations (such as squaring), could belong more in an MPC utility file. This will likely tie into however we end up refactoring mixins and whatnot.